common/components/task-list/template.njk

```diff
@@ -1,29 +1,47 @@
 {% from "govuk/components/tag/macro.njk" import govukTag %}
 
-<ul class="app-task-list">
-  {% for item in params.items %}
-    <li class="app-task-list__item">
-      <span class="app-task-list__task-name">
-        {% if item.href %}
-          <a href="{{ item.href }}" aria-described-by="{{ item.text | urlencode }}-status">
-        {% endif %}
+{% macro taskListItems(items) %}
+  <ul class="app-task-list__items">
+    {% for item in items %}
+      <li class="app-task-list__item">
+        <span class="app-task-list__task-name">
+          {% if item.href %}
+            <a href="{{ item.href }}" aria-described-by="{{ item.text | urlencode }}-status">
+          {% endif %}
+
+          {{ item.html | safe if item.html else item.text }}
 
-        {{ item.html | safe if item.html else item.text }}
+          {% if item.href %}
+            </a>
+          {% endif %}
+        </span>
 
-        {% if item.href %}
-          </a>
+        {% if item.tag %}
+          {{ govukTag({
+            attributes: {
+              id: item.text | urlencode + "-status"
+            },
+            text: item.tag.text,
+            classes: (item.tag.classes or "") + " app-task-list__tag"
+          }) }}
         {% endif %}
-      </span>
+      </li>
+    {% endfor %}
+  </ul>
+{% endmacro %}
+
+{% if params.sections %}
```

Comment:
I wonder if there's a way we can guarantee that both `sections` and `items` isn't set at the same time - perhaps by raising an exception? Then this would be visible in tests where the component is used.

Reply:
I have wondered that in the past for some other similar situations, for example where GOV.UK components use `text` and `html` properties on text elements. Their pattern has been to favour one over the other and to document that in the yaml configuration over throwing an exception.

I've updated the yaml to be even more explicit that `sections` is favoured over `items` when both are provided so hopefully that will help.